tools/pubsys/src/repo/validate_repo/mod.rs

```diff
@@ -122,12 +122,24 @@ fn validate_repo(
 
 /// Common entrypoint from main()
 pub(crate) fn run(args: &Args, validate_repo_args: &ValidateRepoArgs) -> Result<(), Error> {
-    info!(
-        "Using infra config from path: {}",
-        args.infra_config_path.display()
-    );
-    let infra_config =
-        InfraConfig::from_path(&args.infra_config_path).context(repo_error::Config)?;
+    // If a lock file exists, use that, otherwise use Infra.toml
+    let lock_path = &args
+        .infra_config_path
+        .parent()
+        .context(repo_error::Parent {
+            path: &args.infra_config_path,
+        })?
+        .join("Infra.lock");
+    let infra_config = if lock_path.is_file() {
+        info!("Using infra lock from path: {}", lock_path.display());
+        InfraConfig::from_lock_path(lock_path).context(repo_error::Config)?
+    } else {
+        info!(
+            "Using infra config from path: {}",
+            args.infra_config_path.display()
+        );
+        InfraConfig::from_path(&args.infra_config_path).context(repo_error::Config)?
+    };
```

Comment:
These seems to duplicate some code in `upload_ova/mod.rs`. Can a function be extracted? If the error type makes this difficult I can probably help solve that.